### PR TITLE
[SIDE-925] Add .Net Documents

### PR DESCRIPTION
**Pull request by richseviora**
This PR:

- Documents the .Net LDK files.
- Updates the node documentation

Note: I tried to leverage docfx to generate a documentation site. It kept on generating errors so I'm merging this in without that capability yet.

### CODE REVIEW

**Review by bbrewer-oliveai — APPROVED** [review]
Looks great though I defer to your C# and XML expertise here.

File: ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Filesystem/IFilesystemService.cs [hunk 1]
```diff
@@ -5,36 +5,125 @@
...
         public string Name;
+
+        /// <summary>
+        /// The size of the file, in bytes.
+        /// </summary>
         public long Size;
+
+        /// <summary>
+        /// The file permissions (if applicable).
+        /// TODO: Determine what values are sent for each of MacOS and Windows.
```
Comment (BrettMugglinOlive) [inline ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Filesystem/IFilesystemService.cs:63]: Is this a TODO for later?

File: ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/IStreamingCall.cs [hunk 2]
```diff
@@ -3,10 +3,36 @@
 
 namespace OliveHelpsLDK
 {
+    /// <summary>
+    /// The IStreamingCall interface represents an iterable call that users can iterate over.
+    /// Call <see cref="IStreamingCall{TOutput}.MoveNext">MoveNext</see> to begin accessing values.
+    /// .
```
Comment (BrettMugglinOlive) [inline ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/IStreamingCall.cs:9]: I think this is an extra "."

File: ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Process/IProcessService.cs [hunk 3]
```diff
@@ -3,30 +3,81 @@
...
         IStreamingCall<ProcessEvent> Stream(CancellationToken cancellationToken = default);
     }
 
+    /// <summary>
+    /// Contains information about the process.
+    /// </summary>
     public struct ProcessInfo
     {
+        /// <summary>
+        /// The Process ID
```
Comment (BrettMugglinOlive) [inline ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Process/IProcessService.cs:32]: Missing a period

File: ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/FormInputs.cs [hunk 4]
```diff
@@ -217,20 +205,30 @@ public WhisperFormInput.Types.Select ToProto()
...
-        public int Order { get; set; }
-
+        /// <summary>
+        /// Initial value.
+        /// </summary>
         public string Value { get; set; }
 
+        /// <summary>
+        /// The pattern of the field to be populated.
+        /// TODO: Provide example.
```
Comment (BrettMugglinOlive) [inline ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/FormInputs.cs:227]: Same as other TODO

File: ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/Forms.cs [hunk 5]
```diff
@@ -4,22 +4,54 @@
...
     }
 
+    /// <summary>
+    /// Represents the final submission of the form.
+    /// </summary>
     public struct WhisperResult : IWhisperFormResponse
     {
+        /// <summary>
+        /// Whether the form was submitted (<c>true</c>), or rejected/dismissed (<c>false</c>).
+        /// TODO: Confirm that dismissed forms return false.
```
Comment (BrettMugglinOlive) [inline ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/Forms.cs:38]: Has this been confirmed?

File: ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/List.cs [hunk 6]
```diff
@@ -1,44 +1,130 @@
...
+    /// </summary>
     public enum Style
     {
+        /// <summary>
+        /// Standard styling.
+        /// </summary>
         None,
+
+        /// <summary>
+        /// TODO: Find this out.
```
Comment (BrettMugglinOlive) [inline ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/List.cs:31]: A few TODOs here

File: ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Filesystem/IFilesystemService.cs (hunk 1 above)
Comment (richseviora) [reply ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Filesystem/IFilesystemService.cs:202]: It was! We're sending over the Go FileMode format, so I'm linking to that

File: ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/FormInputs.cs (hunk 4 above)
Comment (richseviora) [reply ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/FormInputs.cs:230]: These values are formatted as per HTML Input elements so linking to MDN

File: ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/Forms.cs (hunk 5 above)
Comment (richseviora) [reply ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/Forms.cs:38]: There is no handling for dismissed whispers. I've opened SIDE-1355 to define it.

File: ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/List.cs (hunk 6 above)
Comment (richseviora) [reply ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/List.cs:31]: Addressed!

**Review by DanielWagersOlive — APPROVED** [review]

**Pull request merged**